lib/pbench/server/database/models/users.py

```diff
@@ -74,22 +73,27 @@ class User(Database.Base):
     @property
     def roles(self):
         if self._roles:
-            return [x for x in self._roles.split(";")]
+            return self._roles.split(";")
         else:
             return []
 
     @roles.setter
     def roles(self, value):
-        if isinstance(value, list):
+        try:
             self._roles = ";".join(value)
-        else:
-            raise UserSqlError("Adding role", value, "Value is not a list")
+        except Exception as e:
+            raise UserSqlError("Adding role", value, str(e)) from e
```

Comment:
This is really "Setting role", not "Adding". Not sure the `from e` is likely to be useful here as it's a single statement scope and you're including the text ... but that's fine.